CMakeLists.txt

```diff
@@ -38,9 +38,11 @@ elseif(APPLE)
     add_definitions(-DAPPLE)
     set(CMAKE_INCLUDE_PATH "/usr/local/opt/openssl@1.1/include")
     set(CMAKE_LIBRARY_PATH "/usr/local/opt/openssl@1.1/lib")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -pedantic -Wuninitialized -Winit-self -Weffc++")
 elseif(UNIX)
     add_definitions(-DUNIX)
     add_definitions(-DWNCK_I_KNOW_THIS_IS_UNSTABLE)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -pedantic -Wuninitialized -Winit-self -Weffc++")
```

Comment:
Put all flags in a separate variable and set `CMAKE_CXX_FLAGS ` only once, it'll make things easier going forward (Debug/Release for example).